App/Containers/ThreadsList.js

```diff
@@ -86,14 +94,21 @@ class ThreadsList extends React.PureComponent {
       })
     }
     if (
+      this.props.threads &&
+      prevProps.threads && 
       this.props.threads.length !== prevProps.threads.length
     ) {
       this._notificationPrompt()
     }
   }
 
   _notificationPrompt () {
-    if (this.props.notificationsPrompt && !this.props.tourScreen && this.props.threads.length > 0) {
+    if (
+      this.props.notificationsPrompt &&
+      !this.props.tourScreen &&
+      this.props.threads &&
```

Comment:
Again, check for empty object and empty array.

Reply:
For this case as well as `renderPlaceholder` and `renderList`, it might be a lot cleaner to just have props on this class like `showNotification: boolean`, `showPlaceholder: boolean` (you probably don't need `showList` because it's the opposite of `showPlaceholder`?). Anyway, all the logic to calculate those would be in `mapStateToProps` and not in the UI. Then the `render` function could just use simple logic, checking those props, to decide what to render.

Reply:
Yes good idea, I've now done that for both views, much cleaner. Takes the logic out of the render functions and moves it to `mapStateToProps` as you suggested.